enhancements/cluster-logging/forward_to_cloudwatch.md

```diff
@@ -0,0 +1,198 @@
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+[Cloudwatch][concepts] defines Log Groups and Log Streams.
+The CRD above creates 3 Log Groups named "application", "infrastructure" and "audit".
+A new Log Stream is created for each container producing logs.
+
+**Note**: The installer UI (Addon or OLM) can get AWS credentials from a `cloudcredential.openshift.io/v1`.
+The user only has to provide a `region` to enable cloudwatch forwarding for a cluster.
+Details are out of scope for this proposal.
+
+#### I want to define my own log groups
+
+There are two ways to define custom log groups:
+- Use a Kubernetes label with the group name.
+- Define the group name in the `ClusterLogForwarder.Output`
+
+1. Using a kubernetes label named `LogGroup`
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+spec:
+  outputs:
+   - name: CloudwatchOut
+     type: cloudwatch
+     cloudwatch:
+       region: myregion
+	   groupLabel: LogGroup
+     secret:
+        name: mysecret
+  pipelines:
+  - inputRefs: [application, infrastructure, audit]
+  - outputRefs: [CloudwatchOut]
+```
+
+2. Defining a group in the output spec and use input selectors:
+
+```
+apiVersion: "logging.openshift.io/v1"
+kind: "ClusterLogForwarder"
+  inputs:
+  - name: FooIn
+    application:
+	  namespaces: [foo]
+  - name: BarIn
+    application:
+	  namespaces: [bar]
+  outputs:
+   - name: FooOut
+     type: cloudwatch
+     cloudwatch:
+	   group: foo
+   - name: BarOut
+     type: cloudwatch
+     cloudwatch:
+	   group: bar
+  pipelines:
+  - inputRefs: [FooIn]
+    outputRefs: [FooOut]
+  - inputRefs: [BarIn]
+    outputRefs: [BarOut]
+```
+
+#### I want to use EKS authentication with Cloudwatch
+
+TODO: is this a requirement? If so define an appropriate set of `secret` keys.
+
+### Implementation Details
+
+Using the [fluentd cloudwatch plugin][plugin].
+Most of the CRD fields have an obvious corresponding fluentd parameter.
+
+Set `auto_create_stream true` to create streams and groups on the fly.
+
+`log-stream-name` is set to the `containerID` of the log record.
+This matches the intended use of logs streams, from the [Cloudwatch documentation][concepts]:
+
+> More specifically, a log stream is generally intended to represent the sequence of events coming from the application instance or resource being monitored.
+> For example, a log stream may be associated with an Apache access log on a specific host.
+
+To group logs from multiple containers (by namespace, label etc.) use *Log Groups* as shown in the User Stories section above.
+
+### Open Questions
+
+Should the cluster logging operator need to delete log streams that are no longer in use? For example when containers are deleted?
+
+### Risks and Mitigations
+
+None in particular.
```

Comment:
Risks should include that quota in CW can be exceeded if insufficiently granular streams are configured with CW.
See PutLogEvents on https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/cloudwatch_limits_cwl.html:
```
5 requests per second per log stream. Additional requests are throttled. This quota can't be changed.

The maximum batch size of a PutLogEvents request is 1MB.

800 transactions per second per account per Region, except for the following Regions where the quota is 1500 transactions per second per account per Region: US East (N. Virginia), US West (Oregon), and Europe (Ireland). You can request a quota increase.
```